### PR TITLE
Fixes #11857: Create hook ncf methods that can be used in pre-/post-run hooks

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/11857

### CODE REVIEW

File: tree/20_cfe_basics/log_rudder.cf [hunk 1]
```diff
@@ -142,6 +142,136 @@ bundle agent _rudder_common_reports_generic(technique_name, class_prefix, identi
         ifvarclass => "${class_prefix}_repaired";
 }
 
+
+######################################################################
+##### Reporting for the pre/post hooks
+##### These methods should not be used for normal generic method usage
+##### because they do not respect the stop on repair for audit mode
+######################################################################
```
Comment (amousset) [inline tree/20_cfe_basics/log_rudder.cf]: Could you add a WARN on standard methods to notify these should be ketp in sync?

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
PR rebased

File: tree/20_cfe_basics/log_rudder.cf (hunk 1 above)
Comment (ncharles) [reply tree/20_cfe_basics/log_rudder.cf]: Done

**Pull request merged**